api/desecapi/serializers.py

```diff
@@ -223,69 +223,69 @@ def create(self, validated_data):
         return rrset
 
     def update(self, instance: RRset, validated_data):
-        # TODO this is duplicate code from DRF -- what is this for?
-        #  probably related to bulk serializers!
-        raise_errors_on_nested_writes('update', self, validated_data)
-
         rrs_data = validated_data.pop('records', None)
         if rrs_data is not None:
             instance.set_all_record_contents(set(rrs_data))
 
         ttl = validated_data.pop('ttl', None)
         if ttl and instance.ttl != ttl:
             instance.ttl = ttl
-            instance.save(update_fields={'ttl'})  # TODO is passing ttl needed here?
+            instance.save()
 
         return instance
 
 
 class RRsetListSerializer(ListSerializer):
 
     def to_internal_value(self, data):
-        # TODO this is mostly copy-and-paste from rest_framework.serializers.ListSerializer.to_internal_value!
 
-        # TODO needed?
-        # if html.is_html_input(data):
-        #     data = html.parse_html_list(data, default=[])
+        def key(data_item):
+            return data_item.get('subname', ''), data_item.get('type', None)
+
+        def conflicting_items(current_item):
+            return {
+                other_idx: other_item
```

Comment:
I think the right-hand side is not used here, i.e. this dict can be replaced by a list of indices only.

Reply:
Also, let's be conscious about the decision to introduce an algorithm with quadratic runtime here. (Just sayin', there will be requests which have a few thousand RRs, and we're paying quadratic price just for the convenience of listing the indexes of duplicates.)

One can do this in linear time time by creating an index with key subname/type and the value being a list of indices. The lookup time is then independent of the number of entries.